### PR TITLE
Naming conventions for ASB

### PR DESCRIPTION
**Pull request by SeanFeldman**
Naming conventions for ASB that were made public with NSB.ASB release 6.4.0 and needed to be documented.

@Particular/azure @Particular/documentation please review

TODO:
When doco is merged, issue https://github.com/Particular/NServiceBus.AzureServiceBus/issues/84 should be updated with the doco link

### CODE REVIEW

File: Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        public void Snippet()
+        {
+            #region ASB-NamingConventions-entity-sanitization 6.4
+
+            NamingConventions.EntitySanitizationConvention = (entityName, entityType) =>
+            {
+                if (entityType == EntityType.Queue || entityType == EntityType.Topic)
+                {
+                    var regexQueueAndTopicValidCharacters = new Regex(@"[^a-zA-Z0-9\-\._\/]");
```
Comment (SimonCropp) [inline Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs]: use explicit types not var

File: Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                }
+
+                // Subscription
+                var regexSubscriptionValidCharacters = new Regex(@"[^a-zA-Z0-9\-\._]");
+                return regexSubscriptionValidCharacters.Replace(entityName, "");
+            };
+
+            #endregion
+          
+            Func<ReadOnlySettings, Type, string, bool, string> DetermineQueueNameUsingCustomLogic = (settings, messagetype, queueName, doNotIndividualize) => "";
```
Comment (SimonCropp) [inline Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs]: isnt assigning these values redundant? they are only placeholders

Comment (SeanFeldman) [inline Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs]: Correct, the focus is to document the API, not how to build naming convention for a custom topology.

File: Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs (hunk 1 above)
Comment (SeanFeldman) [inline Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs]: Done. @SimonCropp should this be a compilation error in the snippets project (R# enforced)?

File: Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs (hunk 2 above)
Comment (SimonCropp) [inline Snippets/Snippets_5/Azure/Transports/AzureServiceBus/NamingConventionsUsage.cs]: i will look into this

File: nservicebus/azure/azure-servicebus-transport.md [hunk 3]
```diff
@@ -93,6 +93,22 @@ Incoming message:
...
+
+snippet: ASB-NamingConventions-entity-sanitization
+
+This sanitization allows forward slash `/` in queue and topic names unlike default sanitization convention used by Azure Service Bus transport.
+
+Entities creation:
+
+snippet: ASB-NamingConventions-entity-creation-conventions
+
+WARN: This is an advanced topic and requires full understanding of the topology.
```
Comment (SimonCropp) [inline nservicebus/azure/azure-servicebus-transport.md]: `WARNING` not `WARN`

File: nservicebus/azure/azure-servicebus-transport.md [hunk 4]
```diff
@@ -93,6 +93,22 @@ Incoming message:
 
 snippet: ASB-incoming-message-convention
 
+### Naming Conventions
+
+To have a fine-grained control over entity names generated by the ASB transport, `NamingConventions` class exposes several conventions that can be overwritten to provide customization.
```
Comment (SimonCropp) [inline nservicebus/azure/azure-servicebus-transport.md]: > have a fine-grained control over

Isnt "a" redundant?

Comment (SeanFeldman) [inline nservicebus/azure/azure-servicebus-transport.md]: Frankly, not sure. Checked with Grammerly, it doesn't complaint :)

**Pull request merged**